spikeinterface/comparison/addtemplatesrecording.py

```diff
@@ -0,0 +1,130 @@
+            self.add_recording_segment(recording_segment)
+
+
+
+class AddTemplatesRecordingSegment(BaseRecordingSegment):
+    """
+    TODO
+    """
+
+    def __init__(self, sampling_frequency: float, t_start: int, dtype, spike_vector: np.ndarray, templates: np.ndarray, nbefore: List[int],
+                 amplitude_factor: List[List[float]], target_recording: Union[BaseRecordingSegment, None] = None, t_max: Union[int, None] = None) -> None:
+
+        BaseRecordingSegment.__init__(self, sampling_frequency, t_start=t_start)
+        assert not (target_recording is None and t_max is None)
+
+        self.dtype = dtype
+        self.spike_vector = spike_vector
+        self.templates = templates
+        self.nbefore = nbefore
+        self.amplitude_factor = amplitude_factor
+        self.parent_recording = target_recording
+        self.t_max = target_recording.get_num_frames() if t_max is None else t_max
+
+    def get_traces(self, start_frame: Union[int, None] = None, end_frame: Union[int, None] = None,
+                   channel_indices: Union[List, None] = None) -> np.ndarray:
+        n_channels = self.templates.shape[2]
+        start_frame = 0 if start_frame is None else start_frame
+        end_frame = self.t_max if end_frame is None else end_frame
+
+        if self.parent_recording is not None:
+            traces = self.parent_recording.get_traces(start_frame, end_frame)
+        else:
+            traces = np.zeros([self.t_max, n_channels], dtype=self.dtype)
+        
+        channel_indices = list(range(n_channels)) if channel_indices is None else channel_indices
+
+        mask = (self.spike_vector['sample_ind'] > start_frame - self.templates.shape[1]) & \
+               (self.spike_vector['sample_ind'] < end_frame + self.templates.shape[1])
+        spike_vector = self.spike_vector[mask]
+
+        for spike in spike_vector:
+            t = spike['sample_ind']
+            unit_ind = spike['unit_ind']
+            # spike_nb = np.argmax(self.spike_vector[self.spike_vector['unit_ind'] == unit_ind]['sample_ind'] == t)  # TODO: Very slow
+            template = self.templates[unit_ind, :, channel_indices]
+
+            start_traces = t - self.nbefore[unit_ind] - start_frame
+            end_traces = start_traces + template.shape[0]
+            if start_traces >= end_frame-start_frame or end_traces <= 0:
+                continue
+
+            start_template = 0
+            end_template = template.shape[0]
+
+            if start_traces < 0:
+                start_template = -start_traces
+                start_traces = 0
+            if end_traces > end_frame - start_frame:
+                end_template = template.shape[0] + end_frame - start_frame - end_traces
+                end_traces = end_frame - start_frame
+
+            traces[start_traces : end_traces] += template[start_template : end_template]
```

Comment:
why did you remove the amplitude factor ?

Reply:
Done.